x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/UpdateConnectorSyncJobIngestionStatsAction.java

```diff
@@ -66,6 +67,7 @@ public Request(StreamInput in) throws IOException {
             this.indexedDocumentVolume = in.readLong();
             this.totalDocumentCount = in.readOptionalLong();
             this.lastSeen = in.readOptionalInstant();
+            this.metadata = in.readGenericValue();
```

Comment:
A lot of the usages I see for `in.readGenericValue()` us casting. Like 
```
this.metadata = (Map<String, Object>) in.readGenericValue();
```

Reply:
or even better `in.readGenericMap()`!